src/main/java/net/dv8tion/jda/api/interactions/components/text/Modal.java

```diff
@@ -0,0 +1,214 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.interactions.components.text;
+
+import net.dv8tion.jda.api.interactions.components.ActionComponent;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.internal.interactions.component.ModalImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a Discord Modal
+ *
+ * Replying to an interaction with a modal will cause a form window to pop up on the User's client.
+ *
+ * <b>Only a maximum of 5 components can be included in a Modal, and only {@link net.dv8tion.jda.api.interactions.components.text.TextInput TextInputs} are allowed.</b>
+ *
+ * @see net.dv8tion.jda.api.events.interaction.ModalSubmitInteractionEvent
+ */
+public interface Modal extends ActionComponent
+{
+    /**
+     * The custom id of this modal
+     *
+     * @return The custom id of this modal
+     */
+    @Nonnull
+    String getId();
+
+    /**
+     * The title of this modal
+     *
+     * @return The title of this modal
+     */
+    @Nonnull
+    String getTitle();
+
+    /**
+     * A List of {@link net.dv8tion.jda.api.interactions.components.ActionRow ActionRows} that this modal contains.
+     *
+     * @return List of ActionRows
+     */
+    @Nonnull
+    List<ActionRow> getActionRows();
+
+    @Nonnull
+    @Override
+    default ActionComponent withDisabled(boolean disabled)
+    {
+        throw new UnsupportedOperationException("Modals cannot be disabled!");
+    }
+
+    /**
+     * Creates a new Modal.
+     *
+     * @param customId The custom id for this modal
+     *
+     * @throws IllegalArgumentException
+     *         If the provided customId or title are null
+     *
+     * @return {@link Builder Builder} instance to customize this modal further
+     */
+    @Nonnull
+    @CheckReturnValue
+    static Modal.Builder create(@Nonnull String customId, @Nonnull String title)
+    {
+        Checks.notNull(customId, "Custom ID");
+        Checks.notNull(title, "Title");
+        return new Modal.Builder(customId).setTitle(title);
+    }
+
+    class Builder
+    {
+        private String id;
+        private String title;
+        private final List<ActionRow> components = new ArrayList<>();
+
+        protected Builder(@Nonnull String customId)
+        {
+            setId(customId);
+        }
+
+        /**
+         * Sets the custom id for this modal.
+         *
+         * @param customId Custom id
+         *
+         * @return Builder for chaining convenience
```

Comment:
```suggestion
         * @return The same builder instance for chaining
```

To keep it consistent with SelectMenu's builder